validator/src/bootstrap.rs

```diff
@@ -646,10 +649,16 @@ fn get_rpc_nodes(
             bootstrap_config,
         );
         if rpc_peers.is_none() {
+            if get_rpc_peers_timout.elapsed() > GET_RPC_PEERS_TIMEOUT {
+                error!("Unable to find any RPC peers");
+                return vec![];
```

Comment:
should we just exit with config guidance?  good odds they're misconfigured

Reply:
`std::process::exit()` :slightly_smiling_face:

Reply:
We could... but do we also add an exit [here](https://github.com/solana-labs/solana/pull/30277/files#diff-e8a460ab9cc255ea2dc59d277ca568a9b94f7a14fa87703979c43943debad885R697)?

Reply:
> We could... but do we also add an exit [here](https://github.com/solana-labs/solana/pull/30277/files#diff-e8a460ab9cc255ea2dc59d277ca568a9b94f7a14fa87703979c43943debad885R697)?

hmf... i think it's time this method become properly fallible. let's introduce an error type, make the return value `Result<>` and replace all of the logging/empty `Vec`s

Reply:
Sounds good. I can get to this soon